app/upgrades/v16/upgrades.go

```diff
@@ -77,6 +78,12 @@ func CreateUpgradeHandler(
 			return nil, err
 		}
 
+		// Add both MsgExecuteContract and MsgInstantiateContract to the list of allowed messages.
+		hostParams := keepers.ICAHostKeeper.GetParams(ctx)
+		hostParams.AllowMessages = append(hostParams.AllowMessages, sdk.MsgTypeURL(&cosmwasmtypes.MsgExecuteContract{}))
+		hostParams.AllowMessages = append(hostParams.AllowMessages, sdk.MsgTypeURL(&cosmwasmtypes.MsgInstantiateContract{}))
+		keepers.ICAHostKeeper.SetParams(ctx, hostParams)
+
```

Comment:
Would it be also possible to iterate through existing array of params to make sure that these do not pre-exist in the array?
Just want to check this due to {unknown}{unknown} on what happens if we have the same message set two times in the param